sw/device/lib/dif/templates/dif_autogen.c.tpl

```diff
@@ -0,0 +1,211 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is auto-generated.
+
+#include "sw/device/lib/dif/dif_${ip_snake}_autogen.h"
+
+#include "${ip_snake}_regs.h"  // Generated.
+
+/**
+ * Get the corresponding interrupt register bit offset. INTR_STATE,
+ * INTR_ENABLE and INTR_TEST registers have the same bit offsets, so this
+ * routine can be reused.
+ */
+static bool ${ip_snake}_get_irq_bit_index(
+  dif_${ip_snake}_irq_t irq,
+  bitfield_bit32_index_t *index_out) {
+
+  switch (irq) {
+${irq_to_bit_index_cases}
```

Comment:
Mako has some amount of tem,plating ability... so perhaps it's worth rewriting stuff like this as
```
% for case in ${bit_index_cases}:
  case ${case.rubbish}: ...
% endfor
```

I wouldn't worry about indentation bc we can hit these files with clang-format for that.

https://docs.makotemplates.org/en/latest/syntax.html

Reply:
Oh neat, thanks! Fixed.